spring-amqp/src/main/java/org/springframework/amqp/core/Declarables.java

```diff
@@ -59,7 +59,7 @@ public Collection<Declarable> getDeclarables() {
 	 * @return the filtered list.
 	 * @since 2.2
 	 */
-	public <T> List<T> getDeclarablesByType(Class<T> type) {
+	public <T extends Declarable> List<T> getDeclarablesByType(Class<? extends T> type) {
```

Comment:
I see the point in your change, but I wonder how it is a breaking change to accept it in the current point version?
It is definitely will be safe to postpone this change until the next `3.2` version after May, since I see that code like this `getDeclarablesByType(String.class)` would not be compilable any more.

Reply:
Release `3.2` would be perfectly fine.

Reply:
Why just `Class<T>` is not enough?
Why do you do like that `? extends T`?

Reply:
This is to increase usage flexibility.
`? extends T` allows
```
@Override
public Collection<Exchange> getExchanges() {
    return declarables.getDeclarablesByType(TopicExchange.class);
}
```
or
```
List<Queue> queues = declarables.getDeclarablesByType(AnonymousQueue.class);
```